lib/cpp_peripherial.dart

```diff
@@ -0,0 +1,306 @@
+import 'package:universal_chess_driver/central.dart';
+import 'package:universal_chess_driver/peripherial.dart';
+import 'package:universal_chess_driver/peripherial_client.dart';
+import 'package:universal_chess_driver/utils.dart';
+
+class CppPeripheralRound implements PeripherialRound {
+  String? get fen => null;
+  bool? get isSynchronized => null;
+}
+
+class CppPeripherial implements Peripherial {
+  final PeripherialClient _client;
+  final Central _central;
+  late PeripherialState _state;
+  List<String> _features = [];
+  List<String> _variants = [];
+
+  CppPeripherial(this._client, this._central) {
+    _client.recieve().listen(
+        (dataChunks) => onPeripheralCmd(String.fromCharCodes(dataChunks)));
+    transitionTo(new IterableExchangeState(
+        _central.features.iterator,
+        "feature",
+        IterableExchangeState(
+            _central.variants.iterator, "variant", IdleState(), this._variants),
+        this._features));
+  }
+
+  List<String> get features => _features;
+  List<String> get variants => _variants;
+  Central get central => _central;
+  PeripherialRound get round => new CppPeripheralRound();
+
+  void onPeripheralCmd(String cmd) {
+    print("proto: " + "Peripheral: " + cmd);
+    _state.onPeripheralCmd(cmd);
+  }
+
+  @override
+  void onCentralRoundBegin() {
+    _state.onNewGame();
+  }
+
+  @override
+  void onCentralRoundChange() {
+    if (_central.round.lastMove != null)
+      _state.onNewCentralMove(_central.round.lastMove!);
```

Comment:
```
if (!_central.round.lastMove != null)
  return
_state.onCentralRoundChange();
```
Because in future not only `move` will be needed, `time` for example

Reply:
Ore better check:
```
if (!_central.round.lastMove != null)
  return
```
Only in `SynchronizedState`, where it is needed

Reply:
just
```dart
if (_central.round.lastMove == null)
  return

_state.onCentralRoundChange();
```

Reply:
Hmm, not sure if not better to have only forwarding methods calls here